contracts/meta-oracles/HistoricalPriceFeedv2.sol

```diff
@@ -0,0 +1,304 @@
+     * @param  _medianizerAddress         The oracle address to read current price from
+     * @param  _dataDescription           Description of data in Data Bank
+     * @param  _seededValues              Array of previous days' Historical price values to seed
+     *                                    initial values in list. Should NOT contain the current
+     *                                    days price.
+     */
+    constructor(
+        uint256 _updateFrequency,
+        uint256 _updateTolerance,
+        uint256 _maxDataPoints,
+        address _medianizerAddress,
+        string memory _dataDescription,
+        uint256[] memory _seededValues
+    )
+        public
+    {
+        // Set medianizer address, data description, and instantiate medianizer
+        updateFrequency = _updateFrequency;
+        updateTolerance = _updateTolerance;
+        maxDataPoints = _maxDataPoints;
+        dataDescription = _dataDescription;
+        medianizerInstance = IMedian(_medianizerAddress);
+
+        // Create initial values array from _seededValues and current price
+        uint256[] memory initialValues = createInitialValues(_seededValues);
+
+        // Define upper data size limit for linked list and input initial value
+        initialize(
+            historicalPriceData,
+            _maxDataPoints,
+            initialValues[0]
+        );
+
+        // Cycle through input values array (skipping first value used to initialize LinkedList)
+        // and add to historicalPriceData
+        for (uint256 i = 1; i < initialValues.length; i++) {
+            editList(
+                historicalPriceData,
+                initialValues[i]
+            );
+        }
+
+        // Set next available update timestamp
+        nextAvailableUpdate = block.timestamp.add(updateFrequency);
+    }
+
+    /* ============ External ============ */
+
+    /*
+     * Updates linked list with newest data point by querying medianizer. Is eligible to be
+     * called after nextAvailableUpdate timestamp has passed. Because the nextAvailableUpdate occurs
+     * on a predetermined cadence based on the time of deployment, delays in calling poke do not propogate
+     * throughout the whole dataset and the drift caused by previous poke transactions not being mined
+     * exactly on nextAvailableUpdate do not compound as they would if it was required that poke is called
+     * an updateFrequency amount of time after the last poke.
+     *
+     * By way of example, assume updateFrequency of 24 hours and a updateTolerance of 1 hour. At time 1 the
+     * update is missed by one day and when the oracle is finally called the price is 150, the price feed
+     * then linearizes this price to imply a price at t1 equal to 125. Time 2 the update is 10 minutes late but
+     * since it's within the updateTolerance the value isn't linearized. At time 3 everything falls back in line.
+     *
```

Comment:
Might help the reader to reference `linearizePriceUpdate` function to see how the linearization works